### PR TITLE
Fixt test failures

### PR DESCRIPTION
**Pull request by helenren**
For following tests:

testGetSDKInfo

samples.templateapp

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java [hunk 1]
```diff
@@ -203,7 +203,7 @@ public void enableLogout(boolean b) {
     public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
         Log.i("onEditorAction", "view=" + v + " actionId=" + actionId + " event=" + event);
         // Processing the editor action only on key up to avoid sending events like pass code manager unlock twice.
-        if (event != null && event.getAction() == KeyEvent.ACTION_UP) {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: ```event``` should never be ```null``` here, and if it is, it should fall in the ```else``` block like it does now.

File: libs/test/SalesforceHybridTest/src/com/salesforce/androidsdk/phonegap/SDKInfoPluginTest.java [hunk 2]
```diff
@@ -76,9 +79,17 @@ public void testGetSDKInfo() throws NameNotFoundException, JSONException {
 		assertTrue("Wrong bootconfig oauthScopes", sdkInfoOAuthScopes.contains("api"));
 		assertEquals("Wrong bootconfig oauthRedirectURI", bootconfig.getOauthRedirectURI(), sdkInfoBootConfig.getString("oauthRedirectURI"));
 		assertEquals("Wrong bootconfig remoteAccessConsumerKey", bootconfig.getRemoteAccessConsumerKey(), sdkInfoBootConfig.getString("remoteAccessConsumerKey"));
-		assertEquals("Wrong bootconfig androidPushNotificationClientId", bootconfig.getPushNotificationClientId(), sdkInfoBootConfig.getString("androidPushNotificationClientId"));
+        try {
+            sdkInfoBootConfig.getString("androidPushNotificationClientId");
```
Comment (bhariharan) [inline libs/test/SalesforceHybridTest/src/com/salesforce/androidsdk/phonegap/SDKInfoPluginTest.java]: Instead of doing this, let's add a dummy value for ```androidPushNotificationClientId``` for the test project's ```bootconfig.xml```, and revert this change. The test should pass then.

Comment (helenren) [inline libs/test/SalesforceHybridTest/src/com/salesforce/androidsdk/phonegap/SDKInfoPluginTest.java]: yes, if i add a dummy value there, the test can pass, while for that empty id check in the product code, want to use this verification to make sure it works as expected.

Comment (bhariharan) [inline libs/test/SalesforceHybridTest/src/com/salesforce/androidsdk/phonegap/SDKInfoPluginTest.java]: This test is verifying that the value is read if it exists. So adding a dummy push notification ID should be fine.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java (hunk 1 above)
Comment (helenren) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: so after calling setText, call doEditorAction with EditorInfo.IME_ACTION_GO, the event should be set? While with the debug, after  doEditorAction, the even is null, seems like not enter Enter manually after type a passcode there.

Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/PasscodeActivity.java]: Hitting ```ENTER``` manually triggers 2 events - ```ACTION_DOWN``` and then ```ACTION_UP```. We need to simulate ```ACTION_UP``` in the test. Let's revert the changes to this file.

**Comment by bhariharan** [comment]
👍
